db/hardware_test.go

```diff
@@ -0,0 +1,240 @@
+// Package db - hardware tests
+// The following tests validate database functionality, using both network and
+// instance payloads. A postgres instance with the tink database schema is
+// required in order to run the following tests.
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+
+	_ "github.com/lib/pq"
+)
+
+// TestInsertIntoDB : inserts a single record in the hardware database.
+func TestInsertIntoDB(t *testing.T) {
```

Comment:
It feels like this should be named `setupInsertTests` and then `InsertIntoDB` should basically just be calling it. This way the other tests make more sense.

Reply:
It seems weird to have a test called by other tests, I suggest something like:

```go
func insertTestDataIntoDB(t *testing.T) {
	...
	// content of this func
	...
}

func TestInsertIntoDB(t *testing.T) {
	insertTestDataIntoDB(t)
}

func TestGetByMAC(t *testing.T) {
    insertTestDataIntoDB(t)
	// by mac test
}
...
```